chimedb/dataset/insert.py

```diff
@@ -85,7 +85,9 @@ def insert_dataset(ds_id, base_dset, is_root, state, time):
     except Dataset.DoesNotExist:
         if base_dset:
             Dataset.get(Dataset.id == base_dset)
-        state = DatasetState.select(DatasetState.id).get(DatasetState.id == state)
+        state = (
+            DatasetState.select(DatasetState.id).where(DatasetState.id == state).get()
```

Comment:
This is probably worth trying, but this produces the tautological SQL query
```(sql)
SELECT id FROM datasetstate WHERE id=<state>
```
According to https://peewee.readthedocs.io/en/latest/peewee/querying.html#creating-a-new-record you should be able to set the foreign key field to the primary key value and have it do the same thing, which would allow you to skip this query entirely and just set `state=state` below (where the _variable_ `state` is the primary key value).

Reply:
That gives me a 
```
peewee.IntegrityError: (1452, 'Cannot add or update a child row: a foreign key constraint fails (`test`.`dataset`, CONSTRAINT `dataset_ibfk_1` FOREIGN KEY (`state_id`) REFERENCES `datasetstate` (`id`))')
```